--- origsrc/imlib2-1.4.2/src/lib/asm.h	2009-03-29 15:57:22.390625000 -0500
+++ src/imlib2-1.4.2/src/lib/asm.h	2009-03-29 16:11:38.687500000 -0500
@@ -9,7 +9,7 @@
 # define PR_(sym) __##sym
 #endif
 
-#if defined(__GNUC__) && (__GNUC__ >= 4)
+#if defined(__GNUC__) && (__GNUC__ >= 4) && !defined(__CYGWIN__)
 # define HIDDEN_(sym) .hidden PR_(sym)
 #else
 # define HIDDEN_(sym)
--- origsrc/imlib2-1.4.2/src/lib/common.h	2008-08-17 04:39:13.000000000 -0500
+++ src/imlib2-1.4.2/src/lib/common.h	2009-03-29 16:09:04.062500000 -0500
@@ -16,7 +16,7 @@
 #include <sys/types.h>
 #endif
 
-#if defined(__GNUC__) && (__GNUC__ >= 4)
+#if defined(__GNUC__) && (__GNUC__ >= 4) && !defined(__CYGWIN__)
 #define __hidden __attribute__((visibility("hidden")))
 #else
 #define __hidden
